nemoguardrails/types.py

```
@@ -0,0 +1,261 @@
+                func_data = tc.get("function")
+                if func_data is not None:
+                    raw_args = func_data.get("arguments", {})
+                else:
+                    raw_args = tc.get("args", {})
+                    func_data = {"name": tc.get("name", "")}
+
+                if isinstance(raw_args, str):
+                    try:
+                        args_dict = json.loads(raw_args)
+                    except json.JSONDecodeError:
+                        raise ValueError(f"Tool call arguments are not valid JSON: {raw_args!r}")
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)